File: lib/Role.js
//Class created to perform sql operation
import {TableQuery} from "./TableQuery.js";

 class Role extends TableQuery
 {
    
     constructor(){
        super();
         this.select = `SELECT * FROM role;`;
         this.insert = `INSERT INTO role(title, salary, department_id) VALUES (?,?,?);`;
         this.selectById = `SELECT * FROM role WHERE id = ?`;
         this.delete =  `DELETE FROM role WHERE id = ?`;
         
     }    ;
 }

export {Role}

File: lib/Role.cs
//Class created to perform sql operation
const TableQuery = require('./TableQuery);

 Class Role extends TableQuery
 {
     constructor(){
         this.commonSelect =  `SELECT rl.title as 'Job Title', rl.id,dep.name as department ,rl.salary  
                                FROM role rl 
                                left join department dep on dep.id = rl.department_id`;
         this.select = `${this.commonSelect};`;
         this.insert = `INSERT INTO role(title, salary, department_id) VALUES (?,?,?);`;
         this.selectById = `${this.commonSelect} WHERE id = ?`;
         this.delete =  `DELETE FROM role WHERE id = ?`;
     }    
 }

 module.exports = Role;